From 8a356e8fadcb8b162fcdb6cb3904fc661b20f459 Mon Sep 17 00:00:00 2001
From: Shawn Guo <shawn.guo@freescale.com>
Date: Sun, 14 Jul 2013 21:35:49 +0800
Subject: [PATCH 0005/1074] ENGR00269945: reset: add dummy device_reset() for
 !CONFIG_RESET_CONTROLLER build

Add dummy device_reset() function for !CONFIG_RESET_CONTROLLER build,
so that we do not have to add #ifdef CONFIG_RESET_CONTROLLER in every
single client device drivers that call the function.

Signed-off-by: Shawn Guo <shawn.guo@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 include/linux/reset.h |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/include/linux/reset.h b/include/linux/reset.h
index 6082247..2a8e982 100644
--- a/include/linux/reset.h
+++ b/include/linux/reset.h
@@ -12,6 +12,13 @@ struct reset_control *reset_control_get(struct device *dev, const char *id);
 void reset_control_put(struct reset_control *rstc);
 struct reset_control *devm_reset_control_get(struct device *dev, const char *id);
 
+#ifdef CONFIG_RESET_CONTROLLER
 int device_reset(struct device *dev);
+#else
+static inline int device_reset(struct device *dev)
+{
+	return 0;
+}
+#endif /* CONFIG_RESET_CONTROLLER */
 
 #endif
-- 
1.7.5.4

